### DISCUSSION TITLE
Implementation of an InvenTree Webhook for Real-Time Notifications on Part Additions and Modifications

### DISCUSSION THREAD

**Discussion by jirusev** [inventree/InvenTree#6348] [Q&A]
Hello, 

I am currently in the process of integrating a webhook into our system that should be triggered whenever a part is created or updated. My objective is to have a request containing the details of the affected part sent to 'http://my_url:5000/some_endpoint' immediately following the creation or update event.

While I have consulted the "api-doc/#webhook" section of the documentation, I found that the information provided was somewhat limited and lacked the depth I require to complete this task effectively.

Could you please provide me with a more comprehensive guide or examples on how to establish such a webhook? Specifically, I am looking for details on:

- The proper configuration settings for the webhook to listen to part creation and update events.
- The expected format of the request body that will be sent to my specified endpoint.
- Instructions on how to ensure that the webhook is properly secured, particularly in terms of authenticating the incoming requests.
- Any other relevant details or best practices I should follow when setting up and testing this webhook.

Tanks in advance !

**Comment by matmair** [comment]
We only have support for receiving webhooks build in. Sending is not implemented - see #2038

**Comment by wolflu05** [comment]
You can also use a plugin with the [event mixin](https://docs.inventree.org/en/stable/extend/plugins/event/) and manually call your web hook endpoint or directly implement the logic into a plugin.
